### ISSUE TITLE
hammer parser have problems with host info, discussion how to change hammer parser

### ISSUE THREAD

**Issue by tstrych** [SatelliteQE/robottelo#7514]
TLDR: I would like to create some discussion how to change hammer parser ? And after this change all cli tests probably needs to be runned and fixed if the parsed structure will change. 

Imagine this output from host info:
```
input = [
'Subscription Information: ',
'    Release Version:               ',
'    Autoheal:                      true',
'    Registered by Activation Keys: ',
'     1) rafkIs9iuG',
'    System Purpose:                ',
'        Service Level:  ',
]
```

The function [parse_info](https://github.com/SatelliteQE/robottelo/blob/ec7f27d5af391f03b8a962b73d2b86dce720d22c/robottelo/cli/hammer.py#L188) will be called on this input. This input value is just part of output from `Host.info` command.

This will end with:
```
contents[sub_prop][key] = value
TypeError: list indices must be integers or slices, not str
```
This issue cause a lot of problems in our tests. Since `System purpose` fields were added to host info.

What happens inside [parse_info](https://github.com/SatelliteQE/robottelo/blob/ec7f27d5af391f03b8a962b73d2b86dce720d22c/robottelo/cli/hammer.py#L188) function:

Some examples:
Input
```
[
'Subscription Information: ',
'    Release Version:               ',
'    Autoheal:                      true',
'    Registered by Activation Keys: ',
]
```

Output:
`{'subscription-information': {'release-version': '', 'autoheal': 'true', 'registered-by-activation-keys': ''}}`
Looks fine

Then append this line at the end of Input
```
'     1) rafkIs9iuG',
```
Output: `{'subscription-information': ['rafkIs9iuG']}`
Here you can see all registered-by-activation-keys was rewritten by list. 
[Here in code](https://github.com/SatelliteQE/robottelo/blob/ec7f27d5af391f03b8a962b73d2b86dce720d22c/robottelo/cli/hammer.py#L234)

So I assume the problem was there before `System purpose` fields was added, just addition of these fields makes them visible for our tests.
Probably we don't have a test which verifies information about `subscription-information`. Therefore we didn't find this problem sooner.

And finally when we add another line with [indentation level](https://github.com/SatelliteQE/robottelo/blob/ec7f27d5af391f03b8a962b73d2b86dce720d22c/robottelo/cli/hammer.py#L164) = 1.
In our example append this line to Input. `'    System Purpose:                ',`
And output is gonna be `Typeerror` which is upper in this issue. 

I would expect the output to be:

1. ```{'subscription-information': {'release-version': '', 'autoheal': 'true', 'registered-by-activation-keys': ['rafkIs9iuG'], 'system-purpose': {'service-level': ''}}}```

OR

2. ```{'subscription-information': {'release-version': '', 'autoheal': 'true', 'registered-by-activation-keys': {'1': 'rafkIs9iuG'}, 'system-purpose': {'service-level': ''}}}```

OR smth similar.

I already tried to implement the first one. But I did it by moving the AK from indentation level 1 to indentation level two. ( I do not know if this is possible or how to do that) 

Here is my implementation instead of [lines 233-236](https://github.com/SatelliteQE/robottelo/blob/ec7f27d5af391f03b8a962b73d2b86dce720d22c/robottelo/cli/hammer.py#L233-236)
```
                if isinstance(contents[sub_prop], dict) and current_indent_level == 2 and second_level_key and not contents[sub_prop][second_level_key]:
                    contents[sub_prop][second_level_key] = []

                if isinstance(contents[sub_prop], dict) and current_indent_level == 1:
                    contents[sub_prop] = []

                if (current_indent_level == 2 and second_level_key):
                    contents[sub_prop][second_level_key].append(value)
                else:
                    contents[sub_prop].append(value)
```

**Comment by mirekdlugosz** [comment]
It's not strictly related to this discussion, but... is there even a reason why we parse output that is clearly optimized for human-readability instead of just asking hammer to produce yaml or json, which can be parsed unambiguously (and whose parsers we don't have to maintain)?

**Comment by JacobCallahan** [comment]
Hammer can absolutely output this info in json and yaml. This would be a better option for non-trivial output results.

**Comment by tstrych** [comment]
@mirekdlugosz  and @JacobCallahan, but if robotello supports this form of ourput from `.info` command we should fix it and I'm kindly asking you for help. 

If we want to mainly use the json and yaml output format I am okay with that, but that's for another discussion.

**Comment by JacobCallahan** [comment]
@tstrych you should be able to get the raw json output of the info command by passing the output_format argument.
```python
json_info = Host.info(output_format='json')
```

**Comment by tstrych** [comment]
PR was successfully merged, closing
